### PR TITLE
Fix error with calc_network_catchment function when using SpatialPoints

### PR DESCRIPTION
**Pull request by richardellison**
Also fixes issues with non-distance impedance values.

### CODE REVIEW

File: R/RcppExports.R [hunk 1]
```diff
@@ -2,10 +2,10 @@
 # Generator token: 10BE3573-1514-4C36-9D1C-5A225CD40393
 
 coord_matches <- function(sldf, tolval = 0.000) {
-    .Call('stplanr_coord_matches', PACKAGE = 'stplanr', sldf, tolval)
+    .Call('_stplanr_coord_matches', PACKAGE = 'stplanr', sldf, tolval)
 }
 
 join_spatiallines_coords <- function(sldf, startx, starty) {
-    .Call('stplanr_join_spatiallines_coords', PACKAGE = 'stplanr', sldf, startx, starty)
+    .Call('_stplanr_join_spatiallines_coords', PACKAGE = 'stplanr', sldf, startx, starty)
```
Comment (Robinlovelace) [inline R/RcppExports.R]: Ah - is your system in-line with mine now? Remember it was me who was always changing it like this with the `_` prefix.

File: R/catchmentArea.R [hunk 2]
```diff
@@ -98,14 +98,14 @@ calc_catchment <- function(
     targetintersectd <- sp::SpatialPolygonsDataFrame(targetintersectd,
                                                      data=if(length(calccols) == 1){
                                                        setNames(
-                                                         data.frame(x=sum(targetintersect@data[,calccols])),
+                                                         data.frame(x=sum(targetintersect@data[,calccols], na.rm = TRUE)),
                                                          calccols
                                                        )}else{
-                                                         data.frame(as.list(colSums(targetintersect@data[,calccols])))
+                                                         data.frame(as.list(colSums(targetintersect@data[,calccols],na.rm = TRUE)))
```
Comment (Robinlovelace) [inline R/catchmentArea.R]: Would be a little happier if there were spaces after the commas, i.e. (using Ctl-Shift-A in RStudio):

```
data.frame(as.list(colSums(targetintersect@data[, calccols], na.rm = TRUE))) 
```

File: R/RcppExports.R (hunk 1 above)
Comment (richardellison) [reply R/RcppExports.R]: Used a different computer this time, but seems to be. Still trying to work it out!

File: R/catchmentArea.R (hunk 2 above)
Comment (richardellison) [reply R/catchmentArea.R]: I'll add the spaces. Bad habit of mine to revert to no spaces when I have compact code but I also prefer the spaces to be in there.

**Comment by Robinlovelace** [comment]
Looks good. Not sure it really does enhance tmap, but very happy to merge - great work on this!

**Comment by richardellison** [comment]
I think it is more that tmap enhances stplanr, and not the other way round. We could probably remove it entirely, but tmap was still listed as a suggests (which means that tmap must be installed) so travis was failing.

**Comment by Robinlovelace** [comment]
Ah if that's what enhances means I'm all in favour - no need to download the 'kitchen sink' of **tmap** for the functionality in **stplanr**.

**Comment by Robinlovelace** [comment]
Update: yes @richardellison you're right about the meaning of Enhances:

> Enhances: packages that could improve the performance but not strictly required.

Source: [Thomas Cokelaer](http://thomas-cokelaer.info/blog/2013/07/r-package-dependencies-difference-between-imports-depends-suggests-enhances/).

**Pull request merged**